src/plccng/Scanner/matcher/matcher_test.py

```diff
@@ -0,0 +1,33 @@
+from ...load_spec.parse_spec import parse_lexical_spec
+from ...load_spec.structs import Line
+
+from plccng.Scanner.matcher import matcher
+
+def test_rules_attribute():
+    lexical_spec = make_lexical_spec()
+    test_matcher = matcher.Matcher(lexical_spec.ruleList)
+    print(test_matcher.rules)
+
+def test_empty_line():
+    test_matcher = make_matcher()
+    assert "LexError" == test_matcher.match("")
```

Comment:
match() should take a Line and an index and return one of three kinds of objects: LexError, Token, and Skip.

A test would then look something like...

```python
def test_empty_line():
    matcher = make_matcher()
    line = Line(text="", file=None, number=1)
    result = test_matcher.match(line=line, index=0)
    assert result == LexError(line=line, column=1)
```